src/compiler/emitter.ts

```diff
@@ -4649,44 +4636,78 @@ var __param = this.__param || function(index, decorator) { return function (targ
+                ///
+                /// To ensure this is true in cases of modules with no aliases, e.g.: 
+                /// `import "module"` or `<amd-dependency path= "a.css" \>` we need to add these 
+                /// modules to the end of the dependencies list
+
+                let aliasedModuleNames: string[] = [];
+                let unaliasedModuleNames: string[] = [];
+                let importAliaseNames: string[] = [];
                 for (let importNode of externalImports) {
-                    write(", ");
+                    // Find the name of the external module
+                    let externalModuleName = "";
                     let moduleName = getExternalModuleName(importNode);
                     if (moduleName.kind === SyntaxKind.StringLiteral) {
-                        emitLiteral(<LiteralExpression>moduleName);
+                        externalModuleName = getLiteralText(<LiteralExpression>moduleName);
                     }
-                    else {
-                        write("\"\"");
-                    }
-                }
-                for (let amdDependency of node.amdDependencies) {
-                    let text = "\"" + amdDependency.path + "\"";
-                    write(", ");
-                    write(text);
-                }
-                write("], function (require, exports");
-                for (let importNode of externalImports) {
-                    write(", ");
+
+                    // Find the name of the module alais, if there is one
+                    var importAliasName: string;
                     let namespaceDeclaration = getNamespaceDeclarationNode(importNode);
                     if (namespaceDeclaration && !isDefaultImport(importNode)) {
-                        emit(namespaceDeclaration.name);
+                        importAliasName = getSourceTextOfNodeFromSourceFile(currentSourceFile, namespaceDeclaration.name);
                     }
                     else {
-                        write(getGeneratedNameForNode(<ImportDeclaration | ExportDeclaration>importNode));
+                        importAliasName = getGeneratedNameForNode(<ImportDeclaration | ExportDeclaration>importNode);
+                    }
+
+                    if (importAliasName) {
+                        aliasedModuleNames.push(externalModuleName);
+                        importAliaseNames.push(importAliasName);
+                    }
+                    else {
+                        unaliasedModuleNames.push(externalModuleName);
                     }
                 }
+
+                // Fill in amd-dependency tags
                 for (let amdDependency of node.amdDependencies) {
                     if (amdDependency.name) {
-                        write(", ");
-                        write(amdDependency.name);
+                        aliasedModuleNames.push("\"" + amdDependency.path + "\"");

```

Comment:
we have not used them before.. i really do not feel strongly either ways but i would keep them for consistency.